azure-kusto-data/azure/kusto/data/security.py

```diff
@@ -116,32 +150,61 @@ def __init__(self, kcsb):
         aad_authority_uri = os.environ.get("AadAuthorityUri", CLOUD_LOGIN_URL)
         self.authority_uri = aad_authority_uri + authority if aad_authority_uri.endswith("/") else aad_authority_uri + "/" + authority
 
-    def acquire_authorization_header(self):
+    def _build_kusto_authentication_error(self, error: KustoAuthenticationError) -> KustoAuthenticationError:
```

Comment:
should be a `KustoAuthenticationError` ctor / factory method

Reply:
I think this is beyond of the scope of this task, and this PR is very big already... let's do it later

Reply:
you can have a ctor that accepts the helper itself..